core/src/main/java/org/apache/gravitino/catalog/CredentialManager.java

```diff
@@ -38,16 +39,19 @@ public CredentialManager(
   }
 
   public List<Credential> getCredentials(NameIdentifier identifier) {
+    CredentialPrivilege privilege =
+        getCredentialPrivilege(PrincipalUtils.getCurrentUserName(), identifier);
     return doWithCatalog(
         NameIdentifierUtil.getCatalogIdentifier(identifier),
-        c -> getCredentials(c.catalog(), identifier),
+        catalogWrapper ->
+            catalogWrapper.doWithCredentialOps(
+                credentialOps -> credentialOps.getCredentials(identifier, privilege)),
         NoSuchCatalogException.class);
   }
 
-  private List<Credential> getCredentials(BaseCatalog catalog, NameIdentifier identifier) {
-    throw new NotImplementedException(
-        String.format(
-            "Load credentials is not implemented for catalog: %s, identifier: %s",
-            catalog.name(), identifier));
+  @SuppressWarnings("UnusedVariable")
+  private CredentialPrivilege getCredentialPrivilege(String user, NameIdentifier identifier)
+      throws NotAuthorizedException {
+    return CredentialPrivilege.WRITE;
```

Comment:
Why do we return write privilege here?

Reply:
Besides, do we need a `CredentialManager`, or is the class name proper?  Basically, my feeling is that it manages nothing, just a dispatcher.

Reply:
As Gravitino doesn't provide the feature to check which privileges are granted to a user to a specific fileset, we provide a default WRITE privilege to make the access to fileset works well.